changelog.d/12630.misc

```diff
@@ -0,0 +1 @@
+Add `@cancellable` decorator, for use on endpoint methods that can be cancelled when clients disconnect.
```

Comment:
This changeset doesn't add a decorator. If we've already got a newsfragment with this body from a related PR then fair enough; otherwise I'd suggest

```suggestion
Add helper class for testing request cancellation.
```

Reply:
It's part of a larger series of PRs (basically all the commits in #12583) that I wanted to group under the same changelog message.
Except now that you mention it, it doesn't work so well if the review process spans an RC. I think it's best if I give each PR its own changelog entry for now, then we can group them together at RC time?

(note to self: remember to update the changelog for https://github.com/matrix-org/synapse/pull/12587/files)

Reply:
Yeah, the timing with regard to RCs is the only reason I brought this up. I guess one alternative would be to merge your PRs into a feature branch, and then merge that into develop once all are reviewed?

Reply:
I'm tempted to go for the continuous integration approach rather than a feature branch. I'll update the newsfile.

Reply:
> (note to self: remember to update the changelog for https://github.com/matrix-org/synapse/pull/12587/files)

PRed here: https://github.com/matrix-org/synapse/pull/12663